CRM/Contact/DAO/Contact.php

```diff
@@ -833,6 +833,7 @@ public static function &fields() {
             'keyColumn' => 'name',
             'optionEditPath' => 'civicrm/admin/options/languages',
           ],
+          'readonly' => TRUE,
```

Comment:
I'm not sure about  `'preferred_language'`. AFAIK CiviCRM doesn't do anything clever to set this value programatically?